milestones/tests/test_api.py

```diff
@@ -490,6 +518,72 @@ def test_add_course_content_milestone_inactive_to_active(self):
                 self.test_milestone
             )
 
+    def test_add_course_content_milestone_invalid_requirements(self):
+        """ Unit Test: test_add_course_content_milestone_invalid_requirements """
+        with self.assertRaises(exceptions.InvalidRequirementsException):
+            api.add_course_content_milestone(
+                self.test_course_key,
+                self.test_content_key,
+                self.relationship_types['REQUIRES'],
+                self.test_milestone,
+                set()  # Not JSON serializable
+            )
+
+    def test_add_course_content_milestone_active_exists_update_requirements(self):
+        """ Unit Test: test_add_course_content_milestone_active_exists_update_requirement """
+        api.add_course_content_milestone(
+            self.test_course_key,
+            self.test_content_key,
+            self.relationship_types['REQUIRES'],
+            self.test_milestone,
+            {'min_score': 60}
```

Comment:
I think if we went that route we would want to add a requirement_type field to the Requirements model which would require us to update the edx-milestones implementation every time a client wanted to persist a new kind of requirement. To me, the JSON field route is more flexible.